### ISSUE TITLE
UI is blocking when there are many messages in the conversation

### ISSUE THREAD

**Issue by ChrisLuck** [janimo/textsecure-qml#34]
Hi, I noticed that the UI is blocked when the app is receiving messages. On my Nexus4 this usually takes about 5 second per received message. On one hand, this ui blocking is good because i learned how to 'use' it as a progress indicator. Whenever i get the app in the foreground i am trying to scroll. When the scrolling works i know there are no messages. But when the scrolling is blocked i know that i have to wait some seconds to read my new messages. Maybe you should consider inegrating some kind of progress indicator for receive/decrypt and make the UI non-blocking.

**Comment by janimo** [comment]
I agree this is needed, and it is planned for attachments. I did not think it would be noticeable for plain messages. Does it really take 5 seconds per received text message? If so it may be an unrelated bug.

**Comment by ChrisLuck** [comment]
I just received a message containing only one smiley. While the UI was blocked i was able to slowly count from 20 to 38. So i think it took about 18 seconds.

**Comment by ChrisLuck** [comment]
After deleting the lengty conversation, the blocking-time on receive shrinked down to about a second.

**Comment by janimo** [comment]
Thanks, that's good to know. No wonder I could not reproduce this, I only have short conversations.
So it is likely the naive approaches I use to show the conversation without any optimizations that are causing the lag.

**Comment by janimo** [comment]
The qml.Changed() calls scattered in the code seem to cause a lot of overhead.
